synapse/config/emailconfig.py

```diff
@@ -85,8 +85,11 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
         if email_config is None:
             email_config = {}
 
+        self.implicit_tls = email_config.get("implicit_tls", False)
```

Comment:
what happens if `implicit_tls` is true and `enable_tls` is false?

We should catch such nonsensical configurations rather than leaving the user to discover their error on their own.

Reply:
I added a check for this in [862bc52](https://github.com/matrix-org/synapse/pull/13317/commits/862bc5222426c7ca978d2e323abd2ae62c0d7ba4).